Core/StorageCache.swift

```diff
@@ -57,24 +63,29 @@ public class StorageCache: StorageCacheUpdating {
             HTTPSUpgrade.shared.loadData()
             return result
             
-        case .httpsBloomFilterSpec:
-            return false
-            
         case .surrogates:
             return fileStore.persist(data as? Data, forConfiguration: configuration)
             
         case .trackerDataSet:
-            if fileStore.persist(data as? Data, forConfiguration: configuration) {
-                if TrackerDataManager.shared.reload(etag: etag) != .downloaded {
-                    Pixel.fire(pixel: .trackerDataReloadFailed)
-                    return false
-                }
+            if fileStore.persist(data as? Data, forConfiguration: configuration),
+               TrackerDataManager.shared.reload(etag: etag) == .downloaded {
                 return true
+            } else {
+                Pixel.fire(pixel: .trackerDataReloadFailed)
```

Comment:
I'm not sure about this change: previously we tracked when file has been malformed, that change tracks malformed file and issues related to storage (persistence). I don't think we need this, as amount of free space on the device is outside of our control and not actionable - I'd rather see we stick to old behavior here and adapt it to new privacy config.

Reply:
The main reason behind this change is the cyclomatic complexity of the method was getting too high. I can revert it to the old logic but we may need to break each case into its own method? Or perhaps just the more complex ones?

Reply:
Then we definitely should keep old behavior. I'd just silence the warning as I prefer to have this in one block of code.